### DISCUSSION TITLE
TDP new component proposal guidelines

### DISCUSSION THREAD

**Discussion by leopaul36** [TOSIT-IO/TDP#25] [General]
## ✨ TDP component proposal guidelines

Thanks for wanting to contribute to TDP!

If you think a project should be included to TDP, please act according to the following guidelines:

**Before doing any work**, make sure to open a discussion in the [New components](https://github.com/TOSIT-IO/TDP/discussions/new?category=new-components) category of the [TOSIT-IO](https://github.com/TOSIT-IO) organization.

In this discussion, you should implement the following sections:

```
## Project name
### Presentation of the project
### Why should it be included in TDP?
```

**Note:** Please keep in mind that a proposed project should be **Open Source** under a **non-restrictive license** (Apache v2, MIT, etc.).

The discussion will serve as a debate place for wether or not the proposed project will be included to TDP.

**If the proposal is approved** by the TDP core team, you should then work on the implementation:

1. **Fork the repo** of the proposed project, create a special `*-TDP` branch and add some information about:
* How to **build**
* How to **test**
* Which tests are failing and why 

Please see the [Spark branch 3.2-TDP](https://github.com/TOSIT-IO/spark/tree/branch-3.2-TDP/tdp) for a functionnal example.

**Note:** If the project has dependencies on TDP components (in the `pom.xml` for example), the project should be built with the TDP version of these dependencies.

2. Submit a **Pull request** on the Ansible [tdp-collection](https://github.com/TOSIT-IO/tdp-collection) in which we can deploy the proposed component.

The role acceptance criterias are:
- [ ] Same coding style as the other TDP roles
- [ ] The role variables should be in the `tdp_vars_defaults` of the collection
- [ ] The component's dependencies should be defined in a `tdp_lib_dag/component_name.yml` file in the collection
- [ ] Processes should be managed by **systemd**
- [ ] Service install should follow the Linux [Filesystem Hierarchy Standard (FHS)](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard):
  - [ ] Binaries in `/opt/tdp/service`
  - [ ] Logs in `/var/log/service`
  - [ ] Configuration files in `/etc/service/conf`
- [ ] SSL/TLS encryption should be implemented (if applicable)
- [ ] Kerberos authentication should be implemented (if applicable)
- [ ] High availability should be configurable (if applicable)
- [ ] Commits should strictly follow the [Conventional Commit](https://www.conventionalcommits.org/) convention

3. Send a message in the GitHub discussion of the proposed components so we can check your work 🙂.

We are looking forward to receive your contributions!

**Comment by rpignolet** [comment]


**Reply by leopaul36** [reply]
The idea is to add a section to CONTRIBUTING.md that takes you to these guidelines. From here you are in the right place to propose a new component.
